src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2StreamTests.cs

```diff
@@ -837,8 +837,8 @@ public async Task ContentLength_Received_SingleDataFrameOverSize_Reset()
             var expectedError = new Http2StreamErrorException(1, CoreStrings.Http2StreamErrorMoreDataThanLength, Http2ErrorCode.PROTOCOL_ERROR);
 
             Assert.NotNull(thrownEx);
-            Assert.Equal(expectedError.Message, thrownEx.Message);
-            Assert.IsType<Http2StreamErrorException>(thrownEx.InnerException);
+            Assert.Equal(expectedError.Message, thrownEx.InnerException.Message);
+            Assert.IsType<IOException>(thrownEx.InnerException);
```

Comment:
I definitely like this part of the change. If we don't take anything else from this PR, we should still make sure client errors that cause read to throw always result in IOExceptions.

FWIW I consider this case different than cancelling a read due to the configured read rate limit not being met, which instead results in OperationCanceledExceptions.

Reply:
I didn't notice last time that this was asserting the type of the inner exception. Now I'm thinking the current behavior might be fine...